Traduction de optimization/README.md (#53)
elm-france/guide.elm-lang.org: book/optimization/README.md

# Optimization

Il y deux types d'optimisation majeurs en Elm : l'optimisation de performance et l'optimisation de la taille des ressources.

> - **Performance** &mdash; Le DOM est ce qu'il y a de plus lent dans la navigateur, et de loin. J'ai profilé beaucoup d'application Elm afin de les accélérer, mais la plupart des solutions ont un impact limité. Utiliser de meilleures structures de données ? Négligeable. Mettre en cache des resultats de calcul dans le modèle ? Négligeable _et_ le code est pire désormais. Seul l'usage de `Html.Lazy` et `Html.Keyed` a un impact significatif sur les performances en réduisant le nombre d'opérations sur le DOM.

Reviewer: ```suggestion
- **Performance** &mdash; Le DOM est ce qu'il y a de plus lent dans la navigateur, et de loin. J'ai profilé beaucoup d'applications Elm afin de les accélérer, mais la plupart des solutions ont un impact limité. Utiliser de meilleures structures de données ? Négligeable. Mettre en cache des résultats de calcul dans le modèle ? Négligeable _et_, en plus, on obtient un code encore pire qu'avant. Seul l'usage de `Html.Lazy` et `Html.Keyed` a un impact significatif sur les performances en réduisant le nombre d'opérations sur le DOM.
```

Author: `encore pire` c'est pas super comme formulation

Author: sinon `Negigeable _et_ la qualité du code en pâtit` ?

> - **Performance** &mdash; Le DOM est ce qu'il y a de plus lent dans la navigateur, et de loin. J'ai profilé beaucoup d'applications Elm afin de les accélérer, mais la plupart des solutions ont un impact limité. Utiliser de meilleures structures de données ? Négligeable. Mettre en cache des résultats de calcul dans le modèle ? Négligeable _et_ le code est pire désormais. Seul l'usage de `Html.Lazy` et `Html.Keyed` a un impact significatif sur les performances en réduisant le nombre d'opérations sur le DOM.

Author: ```suggestion
- **Performance** &mdash; Le DOM est ce qu'il y a de plus lent dans la navigateur, et de loin. J'ai profilé beaucoup d'applications Elm afin de les accélérer, mais la plupart des solutions ont un impact limité. Utiliser de meilleures structures de données ? Négligeable. Mettre en cache des résultats de calcul dans le modèle ? Négligeable _et_ la qualité du code en pâtit. Seul l'usage de `Html.Lazy` et `Html.Keyed` a un impact significatif sur les performances en réduisant le nombre d'opérations sur le DOM.
```

@vjousse mieux ?

> - **Taille des ressources** &mdash; Le fonctionnement dans un navigateur implique une considération particulière envers les temps de chargement. Plus la taille des ressources est réduite, plus leur chargement sera rapide sur mobile ou avec une connexion internet réduite. Cela aura vraisemblablement plus d'importance que n'importe quelle autre type d'optimisation. Fort heureusement, le compilateur Elm est capable de rendre le code le plus compact possible sans effort supplémentaire du développeur.

Reviewer: ```suggestion
- **Taille des ressources** &mdash; Le fonctionnement dans un navigateur implique de faire particulièrement attention aux temps de chargement. Plus la taille des ressources est réduite, plus leur chargement sera rapide sur mobile ou avec une connexion internet réduite. Cela aura vraisemblablement plus d'importance que n'importe quel autre type d'optimisation. Fort heureusement, le compilateur Elm est capable de rendre le code le plus compact possible sans effort supplémentaire du développeur.
```
